korge/src/commonMain/kotlin/com/soywiz/korge3d/Shapes3D.kt

```diff
@@ -1,15 +1,47 @@
 package com.soywiz.korge3d
 
-import com.soywiz.korge3d.internal.*
-import com.soywiz.korma.geom.*
+import com.soywiz.korge3d.internal.vector3DTemps
+import com.soywiz.korma.geom.Matrix3D
+import com.soywiz.korma.geom.Vector3D
+import com.soywiz.korma.geom.scale
+
+@Korge3DExperimental
+fun Container3D.shape3D(width: Double=1.0, height: Double=1.0, depth: Double=1.0, drawCommands: MeshBuilder3D.() -> Unit): Shape3D {
+   return  Shape3D(width, height, depth, drawCommands).addTo(this)
+}
+
+/*
+ * Note: To draw solid quads, you can use [Bitmaps.white] + [AgBitmapTextureManager] as texture and the [colorMul] as quad color.
+ */
+@Korge3DExperimental
+class Shape3D(
+    initWidth: Double, initHeight: Double, initDepth: Double,
+    drawCommands: MeshBuilder3D.() -> Unit
+) : ViewWithMesh3D(createMesh(drawCommands).copy()) {
+
+    var width: Double = initWidth
+    var height: Double = initHeight
+    var depth: Double = initDepth
+
+    override fun prepareExtraModelMatrix(mat: Matrix3D) {
+        mat.identity().scale(width, height, depth)
+    }
+
+    companion object {
+
+        fun createMesh(drawCommands: MeshBuilder3D.() -> Unit) = MeshBuilder3D {
+            drawCommands()
+        }
+    }
+}
+
 
 @Korge3DExperimental
 @Deprecated("Kotlin/Native boxes inline+Number")
-inline fun Container3D.box(width: Number, height: Number, depth: Number, callback: Cube3D.() -> Unit = {}): Cube3D
-    = box(width.toDouble(), height.toDouble(), depth.toDouble(), callback)
+inline fun Container3D.cube(width: Number, height: Number, depth: Number, callback: Cube3D.() -> Unit = {}): Cube3D = cube(width.toDouble(), height.toDouble(), depth.toDouble(), callback)
 
 @Korge3DExperimental
-inline fun Container3D.box(
+inline fun Container3D.cube(
```

Comment:
We have several terms: `box`, `cube` and `cuboid`. It seems that other programs including blender uses cube so the change makes sense. But in the MeshBuilder3D looks like there is a cuboid method. Is that a normal cube/box, or is something else?

Reply:
a cube has width == height == depth.
if they are different it is called a cuboid.
(akin to square and rectangle in 2D)

Reply:
Yaah, that's what I thought. I guess I called `box` (though probably `cuboid` more correct) because of that. But it seems that programs use `cube` for simplicity. Or that initially has all its components to 1,1,1, and you use transforms to deform it converting it into a cuboid.